File: create_admin.py
from django.contrib.auth import get_user_model
from django.db import transaction

User = get_user_model()

def create_admin_user():
    try:
        with transaction.atomic():
            # Check if admin user already exists
            if not User.objects.filter(username='admin').exists():
                admin_user = User.objects.create_user(
                    username='admin',
                    password='admin123',
                    email='admin@ahaar.com',
                    first_name='Admin',
                    last_name='User',
                    role='admin'
                )
                print(f"Admin user created successfully: {admin_user.username}")
            else:
                print("Admin user already exists")
    except Exception as e:
        print(f"Error creating admin user: {str(e)}")

if __name__ == '__main__':
    create_admin_user() 

File: frontend/src/scripts/createAdmin.ts
import axios from 'axios';

const createAdmin = async () => {
    try {
        const adminData = {
            username: 'admin',
            password: 'admin123',
            password2: 'admin123',
            email: 'admin@ahaar.com',
            first_name: 'Admin',
            last_name: 'User',
            role: 'admin'
        };

        const response = await axios.post('http://localhost:8000/api/register/', adminData);
        console.log('Admin user created successfully:', response.data);
    } catch (error: any) {
        console.error('Error creating admin user:', error.response?.data || error.message);
    }
};

createAdmin(); 